Python/ceval.c

```diff
@@ -3290,10 +3287,8 @@ _PyEval_EvalFrameDefault(PyFrameObject *f, int throwflag)
 
         case TARGET(CALL_FUNCTION): {
             PREDICTED(CALL_FUNCTION);
-            PyObject **sp, *res;
-            sp = stack_pointer;
-            res = call_function(&sp, oparg, NULL);
-            stack_pointer = sp;
```

Comment:
These stack pointer assignments look a bit like someone left them for documentation purposes, trying to make it clear what is allowed/supposed to happen with the pointer that is passed down.
I''m not saying that it's wrong to remove these assignments, just that Chesterton's Fence might apply.

Reply:
Those strange redirections were added because somebody wanted to use
```
register PyObject **stack_pointer;
```
and it's illegal to take the address of a `register` variable. But the `register` keyword is obsolete, compilers are now much more clever than they were before. Also, `call_function()` should be inlined and then the pointer isn't even a real pointer.